peg/kotlin_pre.h

```diff
@@ -0,0 +1,77 @@
+/*
+ *   Copyright (c) 2019 Masatake YAMATO
+ *   Copyright (c) 2019 Red Hat, Inc.
+ *
+ *   This source code is released for free distribution under the terms of the
+ *   GNU General Public License version 2 or (at your option) any later version.
+ *
+ *   This module contains macros, data decls and prototypes to generate tags for Kotlin.
+ */
+
+/*
+*   INCLUDE FILES
+*/
+
+#include "entry.h"
+#include "numarray.h"
+#include "kind.h"
+#include "read.h"
+
+
+/*
+*   MACROS
+*/
+#define PCC_GETCHAR(auxil) getcFromKotlinFile(auxil)
+#define PCC_MALLOC(auxil,size) eMalloc(size)
+#define PCC_REALLOC(auxil,ptr,size) eRealloc(ptr,size)
+#define PCC_FREE(auxil,ptr) eFreeNoNullCheck(ptr)
+#define PCC_ERROR(auxil) reportError(auxil)
+
+/*
+*   DATA DECLARATIONS
+*/
+typedef enum {
+    K_PACKAGE,
+    K_IMPORT,
+    K_INTERFACE,
+    K_CLASS,
+    K_OBJECT,
+    K_METHOD,
+    K_TYPEALIAS,
+    K_CONSTANT,
+    K_VARIABLE,
+    K_IGNORE
+} KotlinKind;
+
+static kindDefinition KotlinKinds [] = {
+    { true, 'p', "package", "packages", },
+    { true, 'I', "import", "imports", },
```

Comment:
`import X` doesn't define X. X is defined somewhere. The `import` statement uses it. So we should extract X as a reference tag. I will give you more instructions for making a reference tag later.

Reply:
I've looked at how the reference tags work. It is not possible to determine what kind of object is imported in Kotlin, it can be almost anything: package, class, variable, constant or method. So and I've decided to keep it simple for now and drop the support for imports altogether, at least for now.